pkg/kubelet/metrics/collectors/resource_metrics.go

```diff
@@ -235,13 +243,23 @@ func (rc *resourceMetricsCollector) collectContainerMemoryMetrics(ch chan<- metr
 }
 
 func (rc *resourceMetricsCollector) collectContainerSwapMetrics(ch chan<- metrics.Metric, pod summary.PodStats, s summary.ContainerStats) {
-	if s.Swap == nil || s.Swap.SwapUsageBytes == nil {
+	if s.Swap == nil {
 		return
 	}
 
-	ch <- metrics.NewLazyMetricWithTimestamp(s.Swap.Time.Time,
-		metrics.NewLazyConstMetric(containerSwapUsageDesc, metrics.GaugeValue,
-			float64(*s.Swap.SwapUsageBytes), s.Name, pod.PodRef.Name, pod.PodRef.Namespace))
+	swapUsageBytes := float64(0)
+	if s.Swap.SwapUsageBytes != nil {
+		swapUsageBytes = float64(*s.Swap.SwapUsageBytes)
+		ch <- metrics.NewLazyMetricWithTimestamp(s.Swap.Time.Time,
+			metrics.NewLazyConstMetric(containerSwapUsageDesc, metrics.GaugeValue,
+				swapUsageBytes, s.Name, pod.PodRef.Name, pod.PodRef.Namespace))
```

Comment:
for the pod name, should we follow something similar to what we did here: https://github.com/kubernetes/kubernetes/pull/104484/

Reply:
Thanks @SergeyKanzhelev!
Are you referring to `getPodLabelName()`? If so, would you say it's reasonable to expose and re-use it?

Reply:
Hey @SergeyKanzhelev!

I'm a bit confused, as https://github.com/kubernetes/kubernetes/pull/104484 was merged in the v1.25 timeframe (about 3 years ago) and it seems that since then the `getPodLabelName()` was removed from the codebase. Do you know what's the reason for removing it, or why it wasn't used in this file in the first place?

In any case, I'd be happy to explore this refactoring, but I think it would make sense to do that in a follow-up. Would you agree?

Reply:
oh, I didn't realize that the function was removed since. Sorry I didn't check. As mentioned I didn't mean this as a blocking comment. It mostly came to mind as I remembered discussions on label cardinality being too high.

This PR is lgtm from node perspective

Reply:
No worries, thanks a lot for having a look!
I'd be happy to explore this moving forward :)